src/components/Product/product-details.jsx

```diff
@@ -181,20 +181,29 @@ const ProductDetails = ({ location, match }) => {
                       <Button
                         variant="contained"
                         color="primary"
-                        disabled={loading}
+                        disabled={
+                          loading ||
+                          detailsData.quantityInStock < quantityToBeAdded
+                        }
                         className={classes.button}
                         onClick={e => {
                           e.preventDefault();
-                          mutate({
-                            variables: {
-                              productId,
-                              userId: user.id,
-                              quantity: quantityToBeAdded
-                            }
-                          });
+                          if (detailsData.quantityInStock > quantityToBeAdded) {
```

Comment:
The `add to cart` misbehaviour is due to this line 
```suggestion
                          if (detailsData.quantityInStock >= quantityToBeAdded) {
```

Reply:
What misbehaviour?

Reply:
Try to add to cart when you have only one item in stock

Reply:
A little more `context` on the misbehaviour will have been helpful, but thanks, figured it out and made the change. :)

Reply:
Forgive my previous comment @fisayomi. I did not see the generalized `Add to Cart button no longer works.` on the PR rather than on any particular line. Good catch!